google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/readrows/LargeReadRowsResumptionStrategy.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub.readrows;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.retrying.StreamResumptionStrategy;
+import com.google.api.gax.rpc.ApiException;
+import com.google.bigtable.v2.ReadRowsRequest;
+import com.google.bigtable.v2.ReadRowsRequest.Builder;
+import com.google.bigtable.v2.RowSet;
+import com.google.cloud.bigtable.data.v2.internal.RowSetUtil;
+import com.google.cloud.bigtable.data.v2.models.RowAdapter;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStreamResumptionStrategy;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * An implementation of a {@link StreamResumptionStrategy} for merged rows. This class tracks -
+ *
+ * <ul>
+ *   <li>row key for the last row that was read successfully
+ *   <li>row key for large-row that couldn't be read
+ *   <li>list of all row keys for large-rows
+ * </ul>
+ *
+ * Upon retry this class builds a request to omit the large rows & retry from the last row key that
+ * was successfully read.
+ *
+ * <p>This class is considered an internal implementation detail and not meant to be used by
+ * applications.
+ */
+@InternalApi
+public class LargeReadRowsResumptionStrategy<RowT>
+    extends BigtableStreamResumptionStrategy<ReadRowsRequest, RowT> {
+  private final RowAdapter<RowT> rowAdapter;
+  private ByteString lastSuccessKey = ByteString.EMPTY;
+  // Number of rows processed excluding Marker row.
+  private long numProcessed;
+  private ByteString largeRowKey = ByteString.EMPTY;
+  private final AtomicInteger largeRowsCount = new AtomicInteger(0);
+
+  private ReadRowsRequest originalRequest;
+
+  private ReadRowsRequest lastModifiedRequestOnTimeOfError;
+
+  private RowSet previousFailedRequestRowset = null;
```

Comment:
See my comment in RowSetUtil. We can remove `previousFailedRequestRowset`. Since we're updating lastSuccessKey, we'll always update the request to remove everything before lastSuccessKey first. So all the largeRowKeys we've seen before should be excluded from the next request. 

For example:

Original request, row range [r1, r10]. Large row keys: r3, r7.

First attempt: read everything up to r2 successfully, failed on r3. 
Resumption strategy will return new range (r2, r3), (r3, r10].

Second attempt: read everything up to r6 successfully, failed on r7.
Resumption strategy will use the original request [r1, r10],  and update the range to be (r6, r7), (r7, r10].